--- a/libgloss/libnosys/warning.h
+++ b/libgloss/libnosys/warning.h
@@ -37,8 +37,11 @@
 
 /* A canned warning for sysdeps/stub functions.
    The GNU linker prepends a "warning: " string.  */
-#define stub_warning(name) \
+/* #define stub_warning(name) \
   link_warning (name, \
                 #name " is not implemented and will always fail")
+*/
+// suppress stub implementation warnings
+#define stub_warning(name)
 
 #endif /* __WARNING_H__ */
